src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -75,7 +76,76 @@ internal static int CompareStringIgnoreCaseNonAscii(ref char strA, int lengthA,
             return OrdinalCasing.CompareStringIgnoreCase(ref strA, lengthA, ref strB, lengthB);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool Vector128EqualsIgnoreCaseAscii(Vector128<sbyte> vec1, Vector128<sbyte> vec2)
+        {
+            Debug.Assert(Vector128AllAscii(vec1.AsUInt16()));
+            Debug.Assert(Vector128AllAscii(vec2.AsUInt16()));
+            Vector128<sbyte> tmp1 = Vector128.Create((sbyte)0x3f) + vec1;
```

Comment:
From the posted codegen (x64) these constants aren't hoisted outside the loop.
Should this be done manually or will the JIT be thaught to do this for us?

Reply:
As I commented above: I want to look into doing that in the jit (in case if makes sense) instead of rotating code for better performance (it already better the the scalar version).

Reply:
Ah, that's hidden below the arm64 machine code. Sorry, that I didn't spot this before.

Reply:
Presumably, similar to https://github.com/dotnet/runtime/issues/76781